kombu/pidbox.py

```diff
@@ -227,7 +243,18 @@ def get_queue(self, hostname):
         return Queue('%s.%s.pidbox' % (hostname, self.namespace),
                      exchange=self.exchange,
                      durable=False,
-                     auto_delete=True)
+                     auto_delete=True,
+                     queue_arguments=self._get_queue_arguments())
+
+    def _get_queue_arguments(self, ttl=None, expires=None):
```

Comment:
I'd suggest writing this in a different way, as there's a bit of duplication in these methods.
Currently it would be difficult for a subclass to override how the reply_queue_ttl is set,
and it's difficult to add new arguments.

I'd rather see something like a dispatch table of arguments and their types:

```py

class Node(object):
    queue_arguments = {
        'x-message-ttl': maybe_s_to_ms,
        'x-expires': maybe_s_to_ms,
    }

   def __init__(self, ...):
          self.queue_arguments = self._prepare_queue_arguments()
          self.reply_queue_arguments = self._prepare_reply_queue_arguments()

    def _prepare_queue_arguments(self):
        return self._to_queue_arguments([
              ('x-message-ttl', self.queue_ttl),
              ('x-expires, self.queue_expires),
        ])

    def _prepare_reply_queue_arguments(self):
        return self._to_queue_arguments([
              ('x-message-ttl', self.reply_queue_ttl),
              ('x-expires, self.reply_queue_expires),
        ])

     def _to_queue_arguments(self, args):
         return dictfilter({
             key: self.queue_arguments[key](value)
             for key, value in args
        })

```

I guess now we can argue if this rely belongs in the Queue class.